packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html

```diff
@@ -0,0 +1,149 @@
+  >v>
+</gux-rich-text-editor-beta>
+
+<style
+  onload="(function () {
+  (async () => {
+  const { Editor } = await import('https://esm.sh/@tiptap/core@2.7.4');
+  const StarterKit = (await import('https://esm.sh/@tiptap/starter-kit@2.7.4')).default;
+  const Underline = (await import('https://cdn.jsdelivr.net/npm/@tiptap/extension-underline@2.7.4/+esm')).default;
+    
+// Initialize the editor
+const editor = new Editor({
+  element: document.querySelector('.editorElement'),
+  extensions: [StarterKit, Underline],
+  content: 'Start typing here...',
+  injectCSS: false,
+});
+
+// Map the action types to DOM elements and editor commands
+const actionsConfig = [
+  { action: 'bold', command: 'toggleBold', selector: 'gux-rich-text-editor-action[action=bold]' },
+  { action: 'italic', command: 'toggleItalic', selector: 'gux-rich-text-editor-action[action=italic]' },
+  { action: 'strike', command: 'toggleStrike', selector: 'gux-rich-text-editor-action[action=strike]' },
+  { action: 'underline', command: 'toggleUnderline', selector: 'gux-rich-text-editor-action[action=underline]' },
+  { action: 'codeBlock', command: 'toggleCodeBlock', selector: 'gux-rich-text-editor-action[action=codeblock]' },
+  { action: 'blockquote', command: 'toggleBlockquote', selector: 'gux-rich-text-editor-action[action=blockQuote]' },
+  { action: 'orderedList', command: 'toggleOrderedList', selector: 'gux-rich-text-editor-action[action=orderedList]' },
+  { action: 'bulletList', command: 'toggleBulletList', selector: 'gux-rich-text-editor-action[action=bulletList]' },
+  { action: 'clearFormatting', command: 'unsetAllMarks', selector: 'gux-rich-text-editor-action[action=clearFormatting]' },
+  { action: 'undo', command: 'undo', selector: 'gux-rich-text-editor-action[action=undo]' },
+  { action: 'redo', command: 'redo', selector: 'gux-rich-text-editor-action[action=redo]' },
+  { action: 'delete', command: 'delete', selector: 'gux-rich-text-editor-action[action=delete]' }
+];
+
+// Add event listeners and store action elements
+const actions = actionsConfig.map(({ action, command, selector }) => {
+  const element = document.querySelector(selector);
+
+  // Add click event listener for toggling editor command
+  element.addEventListener('click', () => {
+   command == 'delete' ? editor.commands.deleteNode('paragraph') : editor.chain().focus()[command]().run();
+  });
+
+  return { element, action };
+});
+
+// Function to update the active state of each action
+function updateActionState() {
+  actions.forEach(({ element, action }) => {
+    const isActive = editor.isActive(action);
+    element.isActive = isActive;
+  });
+}
+
+// Add a single event listener for updating action states
+editor.on('transaction', updateActionState);
+editor.on('selectionUpdate', updateActionState);
+
+
+
+})();
+
+  })()"
+>
+  .prosemirror pre {
+    white-space: pre-wrap;
+  }
+
+  .tiptap {
+    font-size: 14px;
```

Comment:
What is this styling is every app going to need something like this?

Reply:
This is tiptap styling but you can also have no styling at all and just have the default editor styles. I have configured it to have the same stylings as the figma designs. Its optional really but might be handy to have it as a template I guess. I could add a comment over it maybe.

Reply:
Should we be including it in a light dom css file that is shipped with the components?

Reply:
Yep this makes sense to me 👍